[error]The process '/bin/chmod' failed with exit code 1

**Describe the Issue**

The workflow job in question builds and deploys a flutter web app to the GitHub pages, on gh-pages branch. I've tried several different variations but cannot figure out how to make it work successfully.  I'm not sure at this point if it is a bug or something I'm misconfiguring? 

I'm currently getting this error and I don't know what to do with it:

```
[error]The process '/bin/chmod' failed with exit code 1
```

**Logs**

https://github.com/thomasmarkiewicz/bodysculpting/runs/579246467?check_suite_focus=true

```

Run JamesIves/github-pages-deploy-action@releases/v3
4s
Deployment Failed ❌
Run JamesIves/github-pages-deploy-action@releases/v3
Checking configuration and starting deployment… 🚦
Deploying using Access Token… 🔑
Configuring git…
Git configured… 🔧
Starting to commit changes…
There is nothing to commit. Exiting early… 📭
##[error]The process '/bin/chmod' failed with exit code 1
Deployment Failed ❌
```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -3,17 +3,18 @@
 When contributing to this repository, please first discuss the change you wish to make via issue,	
 email, or any other method with the owners of this repository before making a change. 	
 
-## Pull Request Best Practices	
+## Before Making a Pull Request	
 
 1. Ensure that you've tested your feature/change yourself. As the primary focus of this project is deployment, providing a link to a deployed repository using your branch is preferred. You can reference the forked action using your GitHub username, for example `yourname/github-pages-deplpy-action@master`.	
 2. Make sure you update the README if you've made a change that requires documentation.	
 3. When making a pull request, highlight any areas that may cause a breaking change so the maintainer can update the version number accordingly on the GitHub marketplace.
 4. Make sure you've formatted and linted your code. You can do this by running `yarn format` and `yarn lint`. 
-5. Fix or add any tests where applicable. You can run `yarn test` to run the suite.
+5. Fix or add any tests where applicable. You can run `yarn test` to run the suite. As this action is small in scope it's important that a high level of test coverage is maintained. All tests are written using Jest.
+6. Ensure all typing is accurate and the action compiles correctly by running `yarn build`.
 
 # Deploying
 
-In order to deploy and test your own fork of this action, you must commit the required `node_modules` dependencies. Be sure to run `nvm use` before installing any dependencies. You can learn more about nvm [here](https://github.com/nvm-sh/nvm/blob/master/README.md).
+In order to deploy and test your own fork of this action, you must commit the `node_modules` dependencies. Be sure to run `nvm use` before installing any dependencies. You can learn more about nvm [here](https://github.com/nvm-sh/nvm/blob/master/README.md).
 
 To do this you can follow the instructions below:
 
@@ -42,4 +43,4 @@ $ git checkout -b branchnamehere
 $ git commit -a -m "prod dependencies"
 ```
 
-The `node_modules` folder should _not_ be included when making a pull request. These are only required for GitHub Actions when it consumes the distribution branch branch, the `dev` branch of the project should be free from any dependencies or lib files. 
+The `node_modules` folder should _not_ be included when making a pull request. These are only required for GitHub Actions when it consumes the distribution branch, the `dev` branch of the project should be free from any dependencies or lib files.

--- a/__tests__/git.test.ts
+++ b/__tests__/git.test.ts
@@ -5,6 +5,7 @@ process.env['GITHUB_SHA'] = '123'
 import {action} from '../src/constants'
 import {deploy, generateBranch, init, switchToBaseBranch} from '../src/git'
 import {execute} from '../src/execute'
+import {rmRF} from '@actions/io'
 
 const originalAction = JSON.stringify(action)
 
@@ -13,6 +14,10 @@ jest.mock('@actions/core', () => ({
   getInput: jest.fn(),
   isDebug: jest.fn(),
   info: jest.fn()
+}))
+
+jest.mock('@actions/io', () => ({
+  rmRF: jest.fn()
 }))
 
 jest.mock('../src/execute', () => ({
@@ -316,7 +321,8 @@ describe('git', () => {
       await deploy(action)
 
       // Includes the call to generateBranch
-      expect(execute).toBeCalledTimes(13)
+      expect(execute).toBeCalledTimes(11)
+      expect(rmRF).toBeCalledTimes(1)
     })
 
     it('should execute commands with single commit toggled', async () => {
@@ -334,7 +340,8 @@ describe('git', () => {
       await deploy(action)
 
       // Includes the call to generateBranch
-      expect(execute).toBeCalledTimes(19)
+      expect(execute).toBeCalledTimes(17)
+      expect(rmRF).toBeCalledTimes(1)
     })
 
     it('should execute commands with clean options, ommits sha commit message', async () => {
@@ -354,7 +361,8 @@ describe('git', () => {
       await deploy(action)
 
       // Includes the call to generateBranch
-      expect(execute).toBeCalledTimes(13)
+      expect(execute).toBeCalledTimes(11)
+      expect(rmRF).toBeCalledTimes(1)
     })
 
     it('should execute commands with clean options stored as an array instead', async () => {
@@ -373,7 +381,8 @@ describe('git', () => {
       await deploy(action)
 
       // Includes the call to generateBranch
-      expect(execute).toBeCalledTimes(13)
+      expect(execute).toBeCalledTimes(11)
+      expect(rmRF).toBeCalledTimes(1)
     })
 
     it('should gracefully handle incorrectly formatted clean exclude items', async () => {
@@ -391,7 +400,8 @@ describe('git', () => {
 
       await deploy(action)
 
-      expect(execute).toBeCalledTimes(13)
+      expect(execute).toBeCalledTimes(11)
+      expect(rmRF).toBeCalledTimes(1)
     })
 
     it('should stop early if there is nothing to commit', async () => {
@@ -407,7 +417,8 @@ describe('git', () => {
       })
 
       await deploy(action)
-      expect(execute).toBeCalledTimes(14)
+      expect(execute).toBeCalledTimes(12)
+      expect(rmRF).toBeCalledTimes(1)
     })
 
     it('should throw an error if one of the required parameters is not available', async () => {
@@ -427,7 +438,8 @@ describe('git', () => {
       try {
         await deploy(action)
       } catch (e) {
-        expect(execute).toBeCalledTimes(2)
+        expect(execute).toBeCalledTimes(0)
+        expect(rmRF).toBeCalledTimes(1)
         expect(e.message).toMatch(
           'The deploy step encountered an error: No deployment token/method was provided. You must provide the action with either a Personal Access Token or the GitHub Token secret in order to deploy. If you wish to use an ssh deploy token then you must set SSH to true. ❌'
         )

--- a/__tests__/main.test.ts
+++ b/__tests__/main.test.ts
@@ -7,12 +7,17 @@ import '../src/main'
 import {action} from '../src/constants'
 import run from '../src/lib'
 import {execute} from '../src/execute'
+import {rmRF} from '@actions/io'
 import {setFailed} from '@actions/core'
 
 const originalAction = JSON.stringify(action)
 
 jest.mock('../src/execute', () => ({
   execute: jest.fn()
+}))
+
+jest.mock('@actions/io', () => ({
+  rmRF: jest.fn()
 }))
 
 jest.mock('@actions/core', () => ({
@@ -42,7 +47,8 @@ describe('main', () => {
       debug: true
     })
     await run(action)
-    expect(execute).toBeCalledTimes(20)
+    expect(execute).toBeCalledTimes(18)
+    expect(rmRF).toBeCalledTimes(1)
   })
 
   it('should throw if an error is encountered', async () => {

--- a/package.json
+++ b/package.json
@@ -35,7 +35,8 @@
   "dependencies": {
     "@actions/core": "1.2.3",
     "@actions/exec": "1.0.3",
-    "@actions/github": "2.1.1"
+    "@actions/github": "2.1.1",
+    "@actions/io": "1.0.2"
   },
   "devDependencies": {
     "@types/jest": "25.2.1",

--- a/src/git.ts
+++ b/src/git.ts
@@ -1,4 +1,5 @@
 import {info} from '@actions/core'
+import {rmRF} from '@actions/io'
 import {ActionInterface} from './constants'
 import {execute} from './execute'
 import {
@@ -238,11 +239,7 @@ export async function deploy(action: ActionInterface): Promise<void> {
       )} ❌`
     )
   } finally {
-    // Ensures the deployment directory is safely removed.
-    await execute(
-      `chmod u+w -R ${temporaryDeploymentDirectory}`,
-      action.workspace
-    )
-    await execute(`rm -rf ${temporaryDeploymentDirectory}`, action.workspace)
+    // Ensures the deployment directory is safely removed after each deployment.
+    await rmRF(temporaryDeploymentDirectory)
   }
 }

